WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/Custom Range Filter/DateRangeFilterViewController.swift

```diff
@@ -0,0 +1,231 @@
+        if startDateExpanded {
+            tempRows.append(.startDatePicker)
+        }
+        tempRows.append(.endDateTitle)
+        if endDateExpanded {
+            tempRows.append(.endDatePicker)
+        }
+        rows = tempRows
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension DateRangeFilterViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension DateRangeFilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row] {
+        case .startDateTitle:
+            startDateExpanded = !startDateExpanded
+            configureRows()
+            tableView.reloadData()
+        case .endDateTitle:
+            endDateExpanded = !endDateExpanded
+            configureRows()
+            tableView.reloadData()
+        default:
+            return
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+}
+
+// MARK: - Cell configuration
+//
+private extension DateRangeFilterViewController {
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as TitleAndValueTableViewCell where row == .startDateTitle:
+            configureStartDateTitle(cell: cell)
+            break
+        case let cell as DatePickerTableViewCell where row == .startDatePicker:
+            configureStartDatePicker(cell: cell)
+            break
+        case let cell as TitleAndValueTableViewCell where row == .endDateTitle:
+            configureEndDateTitle(cell: cell)
+            break
+        case let cell as DatePickerTableViewCell where row == .endDatePicker:
+            configureEndDatePicker(cell: cell)
+            break
+        default:
+            fatalError()
+            break
+        }
+    }
+
+    func configureStartDateTitle(cell: TitleAndValueTableViewCell) {
+        cell.updateUI(title: Localization.startDateTitle, value: startDate?.toString(dateStyle: .medium, timeStyle: .none))
+        cell.selectionStyle = .none
+    }
+
+    func configureStartDatePicker(cell: DatePickerTableViewCell) {
+        if let startDate = startDate {
+            cell.getPicker().setDate(startDate, animated: false)
```

Comment:
I'm guessing setting `animated` has some issues? 😅

Reply:
It's okay actually, but as I expect the date to be already selected when the cell is rendered, I don't want the picker to move with an animation from the current date. If the chosen date and the current date were on different months, reopening the screen you would see the picker change month. It might seem strange.